src/api/Reducers/propertyTree.js

```diff
@@ -68,13 +68,41 @@ const propertyOwners = (state = {}, action = {}) => {
           subowners: owner.subowners,
           tags: owner.tags || []
         };
+        
+        // Ensure the parents of the uri have the links to the new entry
+        // Get parent uri
+        var periodPos = owner.uri.lastIndexOf('.');
+        var parentUri = owner.uri.substring(0, periodPos);
+
+        // If that parent exists and the link doesn't exist, add the link to the parent
+        if (parentUri && newState[parentUri] && !newState[parentUri].subowners.includes(owner.uri)) {
+          newState[parentUri].subowners.push(owner.uri);
+        }
       });
       return newState;
     }
     case actionTypes.removePropertyOwners: {
-      const newState = { ...state };
+      let newState = { ...state };
+      
       action.payload.uris.forEach((uri) => {
+        // Delete this particular property owner
         delete newState[uri];
+
+        // Delete the parent's link to the property owner 
+        var periodPos = uri.lastIndexOf('.');
+        var parentUri = uri.substring(0, periodPos);
+        const index = newState[parentUri].subowners.indexOf(uri);
+        // If found, remove
+        if (index > -1) { 
+          // 2nd parameter means remove one item only
+          newState[parentUri].subowners.splice(index, 1); 
+        }
+
+        // Delete subowners that have been flattened
+        const related = Object.keys(newState).filter(value => value.includes(`${uri}.`));
+        related.forEach((subOwnerUri) => 
+          delete newState[subOwnerUri]
+        );
```

Comment:
```suggestion
        related.forEach((subOwnerUri) => delete newState[subOwnerUri]);
```

Any specific reason for it to not be on one row?